src/Workspaces/Remote/Core/RemoteCallback.cs

```diff
@@ -96,16 +94,12 @@ private static bool ReportUnexpectedException(Exception exception, CancellationT
                     return false;
                 }
 
-                // Log unexpected state where a cancellation exception occurs without being requested.
-                return FatalError.ReportAndCatch(exception);
```

Comment:
📝 This was intentionally added to ensure cancellation NFE was reported in cases where the cancellation exception was truly an error. Previously, we had no insight into this failure.

Reply:
Didn't actually mean to remove this. Reverted.